game/scripts/vscripts/abilities/carapace/boss_carapace_headbutt.lua

```diff
@@ -0,0 +1,131 @@
+LinkLuaModifier("modifier_boss_carapace_headbutt_slow", "abilities/carapace/boss_carapace_headbutt.lua", LUA_MODIFIER_MOTION_NONE)
+
+--------------------------------------------------------------------------------
+
+boss_carapace_headbutt = class(AbilityBaseClass)
+
+--------------------------------------------------------------------------------
+
+function boss_carapace_headbutt:GetPlaybackRateOverride()
+	return 0.25
+end
+
+--------------------------------------------------------------------------------
+
+if IsServer() then
+	function boss_carapace_headbutt:OnAbilityPhaseStart()
+		local caster = self:GetCaster()
+		local width = self:GetSpecialValueFor("width")
+		local damage_range = self:GetSpecialValueFor("damage_range")
+		local castTime = self:GetCastPoint()
+		local direction = caster:GetForwardVector()
+
+		DebugDrawBoxDirection(caster:GetAbsOrigin(), Vector(0,-width,0), Vector(damage_range*2,width,50), direction, Vector(255,0,0), 1, castTime)
+		return true
+	end
+end
+
+--------------------------------------------------------------------------------
+
+if IsServer() then
+	function boss_carapace_headbutt:GetEnemies()
+		local caster = self:GetCaster()
+		local damage_range = self:GetSpecialValueFor("damage_range")
+
+		local startPoint = caster:GetAbsOrigin()
+		local endPoint = caster:GetAbsOrigin() + (caster:GetForwardVector() * damage_range)
+
+		local enemies = FindUnitsInLine(
+			caster:GetTeamNumber(),
+			startPoint,
+			endPoint,
+			caster,
+			self:GetSpecialValueFor("width"),
+			DOTA_UNIT_TARGET_TEAM_ENEMY,
+			DOTA_UNIT_TARGET_ALL,
+			DOTA_UNIT_TARGET_FLAG_NONE)
+
+		return enemies
+	end
+end
+
+--------------------------------------------------------------------------------
+
+if IsServer() then
+	function boss_carapace_headbutt:OnSpellStart()
+		local caster = self:GetCaster()
+		local range = self:GetSpecialValueFor("range")
+		local target = (caster:GetForwardVector() * range) + caster:GetAbsOrigin()
+
+		caster:EmitSound("Hero_Juggernaut.BladeDance")
+
+		local smoke = ParticleManager:CreateParticle("particles/econ/items/antimage/antimage_ti7_golden/antimage_blink_start_ti7_golden_smoke.vpcf", PATTACH_CUSTOMORIGIN, caster)
+		ParticleManager:SetParticleControl(smoke, 0, caster:GetAbsOrigin())
+		ParticleManager:ReleaseParticleIndex(smoke)
+
+		local enemies = self:GetEnemies()
+
+		for k,v in pairs(enemies) do
```

Comment:
avoid using one letter variables and don't name if you are not using, try something more meaningful like:
`for _,victim in pair(enemies) do`